api/v1alpha1/contour_types.go

```diff
@@ -302,6 +317,75 @@ const (
 	GCPLoadBalancerProvider   LoadBalancerProviderType = "GCP"
 )
 
+type AwsLoadBalancerParameters struct {
+	// AllocationID is a list of Allocation IDs of Elastic IP addresses that are
+	// to be assigned to created Network Load Balancer.
+	// Example: "eipalloc-<xxxxxxxxxxxxxxxxx>"
+	//
+	// See: https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html
+	//
+	// +optional
+	AllocationID []string `json:"allocationId,omitempty"`
+}
+
+type AzureLoadBalancerParameters struct {
+	// Address is the desired IP of LoadBalancer. If LoadBalancerScope is Internal,
+	// IP address must reside in same virtual network as AKS and must not already
+	// be assigned to a resource. If IP address does not reside in same subnet as
+	// AKS, subnet parameter is also required.
+	//
+	// See:
+	// 	 https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address
+	// 	 https://docs.microsoft.com/en-us/azure/aks/internal-lb#specify-an-ip-address
+	//
+	// xref: https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha1/gateway_types.go#L492-L498
+	//
+	// +optional
+	Address string `json:"address,omitempty"`
+
+	// ResourceGroup is name of resource group where the desired IP resides. Relevant
+	// only if LoadBalancerScope is External.
+	// Omit if desired IP is created in same resource group as k8s cluster.
+	//
+	// +optional
+	ResourceGroup string `json:"resourceGroup,omitempty"`
+
+	// Subnet is name of subnet resource where the desired IP resides. Relevant
+	// only if LoadBalancerScope is Internal and desired IP does not reside in
```

Comment:
s/Internal/"Internal"/

Reply:
Can you see if we can add kubebuilder validation for the fields you're introducing, i.e. min/max length, etc..

Reply:
I added validations based on naming limits from cloud provider's side.